[quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34629 [quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2**
* #33945 [quant][onnx] Add support to convert max_pool2d quantized pytorch op to C2

Summary:
Add support for sigmoid in the conversion flow through onnx

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_quantized_sigmoid
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_small_model

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20433680](https://our.internmc.facebook.com/intern/diff/D20433680)

supriyar committed:
[quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2

Summary:
Add support for sigmoid in the conversion flow through onnx

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_quantized_sigmoid
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_small_model

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2"


Summary:
Add support for sigmoid in the conversion flow through onnx

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_quantized_sigmoid
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_small_model

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f reviewed (approved):
LGTM

houseroad commented on torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp:
```diff
@@ -79,6 +79,10 @@ double getScaleFromInput(Node* input_node) {
+  } else if (input_name == "aten::sigmoid") {
+    // For the _caffe2::Int8Sigmoid op output scale is 1.0/256
+    // And output zero_point is set to 0 (quint8 type).
+    return 1.0 / 256;
```
Nit: use 1.0L to be safe

houseroad commented on test/onnx/test_pytorch_onnx_caffe2_quantized.py:
```diff
@@ -40,6 +41,30 @@ def forward(self, x):
         x = np.random.random((1, 2)).astype("float32")
         self.generic_test(QModule(op), (x,), input_names=["x"])
 
+    def generic_model_test(self, model, sample_inputs, input_names=None):
```
Can we just extend generic_test?

houseroad reviewed (commented):
Some inline comments

supriyar committed:
Update on "[quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2"


Summary:
Add support for sigmoid in the conversion flow through onnx

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_quantized_sigmoid
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_small_model

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20433680](https://our.internmc.facebook.com/intern/diff/D20433680)

[ghstack-poisoned]

raghuramank100 commented on test/onnx/test_pytorch_onnx_caffe2_quantized.py:
```diff
@@ -144,7 +157,7 @@ def forward(self, x):
 
         x = np.random.rand(1, 2, 3, 4).astype("float32")
-        self.generic_test(QUpsampleModule(), (x,), input_names=["x"])
+        self.generic_test(QUpsampleModule(), (x,), input_names=["x"], decimal=0)
```
Suggest using decimal=5 for cases where we expect an exact match.

supriyar replied:
Oops, yeah my bad

supriyar committed:
Update on "[quant][onnx] Support conversion of quantized sigmoid operator from pytorch to caffe2"


Summary:
Add support for sigmoid in the conversion flow through onnx

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_quantized_sigmoid
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_small_model

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20433680](https://our.internmc.facebook.com/intern/diff/D20433680)

[ghstack-poisoned]
